src/Common/src/Microsoft/Win32/SafeHandles/SafeBrotliHandle.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    internal sealed class SafeBrotliEncoderHandle : SafeHandle
+    {
+        public SafeBrotliEncoderHandle() : base(IntPtr.Zero, true) { }
+
+        protected override bool ReleaseHandle()
+        {
+            if (handle != IntPtr.Zero)
+            {
+                Interop.Brotli.BrotliEncoderDestroyInstance(handle);
+                SetHandle(IntPtr.Zero);
```

Comment:
You do not need to call SetHandle yourself. The caller does it for you.

Reply:
> You do not need to call SetHandle yourself.  The caller does it for you.

You mean the runtime caller of this?  I'm not seeing that.  This:
```C#
using System;
using System.Runtime.InteropServices;

class Program
{
    static void Main()
    {
        var msf = new MySafeHandle((IntPtr)12345);
        Console.WriteLine(msf.DangerousGetHandle());
        msf.Dispose();
        Console.WriteLine(msf.DangerousGetHandle());
    }
}

class MySafeHandle : SafeHandle
{
    public MySafeHandle(IntPtr handle) : base(IntPtr.Zero, ownsHandle: true)
    {
        SetHandle(handle);
    }

    public override bool IsInvalid => handle == IntPtr.Zero;
    protected override bool ReleaseHandle()
    {
        Console.WriteLine("Invoked");
        return true;
    }
}
```
prints:
```
12345
Invoked
12345
```
rather than:
```
12345
Invoked
0
```
(That said, I don't see a need to set the handle to zero here.)

Reply:
You are right - the runtime caller will not do it for you. It just not necessary to set the handle to zero here.

Reply:
> It just not necessary to set the handle to zero here.

Yup, agreed.